### PR TITLE
perf(css/minifier): improve

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Reduce visitors + fix small bug

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_minifier/src/compressor/media.rs [hunk 1]
```diff
@@ -28,6 +28,22 @@ impl Compressor {
         )
     }
 
+    pub(super) fn compress_media_query_list(&mut self, media_query_list: &mut MediaQueryList) {
+        let mut already_seen: Vec<MediaQuery> = Vec::with_capacity(media_query_list.queries.len());
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compressor/media.rs:32]: Please use `dedup` function instead

Comment (kdy1) [reply crates/swc_css_minifier/src/compressor/media.rs:32]: I mean https://github.com/swc-project/swc/blob/c4a7f21a71f619106e3293c8b5121842a13f49c9/crates/swc_css_minifier/src/compressor/selector.rs#L311-L342
Please move `deup` to `util` and use it

File: crates/swc_css_minifier/src/compressor/supports.rs [hunk 2]
```diff
@@ -132,66 +136,83 @@ impl Compressor {
...
-                }
 
-                n.conditions = new_conditions;
+                    n.conditions = new_conditions;
+                }
             }
             _ => {}
         }
+
+        let mut already_seen: Vec<SupportsConditionType> = Vec::with_capacity(n.conditions.len());
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compressor/supports.rs:203]: Same here

**Comment by alexander-akait** [comment]
@kdy1 Fixed :+1:

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_atoms

**Pull request merged**